src/main/java/de/zalando/aruha/nakadi/controller/EventPublishingController.java

```diff
@@ -75,17 +85,16 @@ private ResponseEntity response(final EventPublishResult result) {
         }
     }
 
-    private ResponseEntity doWithMetrics(final String eventTypeName, final long startingNanos,
-                                         final EventProcessingTask task) throws NakadiException {
+    private ResponseEntity doWithTimerMetric(final EventTypeMetrics eventTypeMetrics, final long startingNanos,
```

Comment:
I think the motivation for that is to be able to pass an initial time that is not right after the invocation of this method.

Maybe it would make sense to change it now. Seems a good idea to control time inside of this method.